File: ext/examples/resolution_size.rs
use algebra::module::Module;
use ext::{chain_complex::ChainComplex, utils::query_module};

fn main() -> anyhow::Result<()> {
    ext::utils::init_logging()?;

    let res = query_module(None, false)?;

    for s in (0..res.next_homological_degree()).rev() {
        let module = res.module(s);
        for t in res.min_degree() + s..=module.max_computed_degree() {
            print!("{}, ", module.dimension(t));
        }
        println!();
    }
    Ok(())
}

File: ext_py/examples/resolution_size.py
#!/usr/bin/env python3
"""Print the dimension of each module in the resolution, by homological degree.

Python port of ext/examples/resolution_size.rs.
"""

import _query as query


def main():
    res = query.query_module()

    for s in reversed(range(res.next_homological_degree())):
        module = res.module(s)
        for t in range(res.min_degree() + s, module.max_computed_degree() + 1):
            print(f"{module.dimension(t)}, ", end="")
        print()


if __name__ == "__main__":
    main()
